packages/gatsby/src/internal-plugins/query-runner/page-query-runner.js

```diff
@@ -65,8 +66,13 @@ emitter.on(`DELETE_NODE`, action => {
 })
 
 const runQueuedActions = async () => {
-  if (active) {
-    runQueries()
+  if (active && !running) {
+    try {
+      running = true
+      await runQueries()
+    } finally {
+      running = false
+    }
```

Comment:
This lock has potential of some timing issues - if some node gets updated while queries are running we won't run queries for that updated node - they'll stay in dirty nodes queue until another thing in the future would trigger next query running.

Does running 2 concurrent `runQueries()` without this lock cause problems? (others than unnecessary query running of course)

Reply:
It feels wrong that it runs it twice, but I couldn't find the original source from where it was happening. I can fix that timing issue.